### PR TITLE
doc: improve documentation about release process

### PR DESCRIPTION
**Pull request by monperrus**

### CODE REVIEW

**Comment by monperrus** [comment]
@pvojtechovsky ready for merge

**Comment by pvojtechovsky** [comment]
@monperrus nice document, but I do not understand that much. May be @surli should check and merge it.

File: doc/_release/Release.md [hunk 1]
```diff
@@ -2,49 +2,61 @@
...
 This article is a short summary of the [official documentation of sonatype](http://central.sonatype.org/pages/ossrh-guide.html), an [article by yegor](http://www.yegor256.com/2014/08/19/how-to-release-to-maven-central.html) and [official documentation of maven release plugin](http://maven.apache.org/maven-release/maven-release-plugin/).
 
-## Main Workflow
-1. open an account on `https://gforge.inria.fr`, add an SSH key there, check that you are a member of project `spoon`
+## Core tasks:
+
+* [ ] Release on Maven Central
+* [ ] Release on Github
+* [ ] PR to update the information: #2648
+* [ ] PR to remove deprecated methods: #2649
```
Comment (surli) [inline doc/_release/Release.md]: I think you copy/paste a bit fast: you should remove the references to old PRs

File: doc/_release/Release.md [hunk 2]
```diff
@@ -2,49 +2,61 @@
...
+    * Sonatype username and password
+    * GPG keyname and passphrase
+
+## Maven Release
+
 1. clean your project for the release and prepare the release `mvn release:clean release:prepare`
 1. `mvn release:perform` (sends the new version on Maven Central)
 1. check that the new version is on Maven Central (connect to `oss.sonatype.org`)
-1. push the release commit on Github (git push origin master)
+1. alternatively `mvn -Prelease deploy -DskipTests`
```
Comment (surli) [inline doc/_release/Release.md]: Why that? A deploy with skipping test? A bit ugly isn't it?

File: doc/_release/Release.md [hunk 3]
```diff
@@ -81,18 +93,15 @@ To push your archive on Maven Central, you must sign before your jar with GPG, a
...
 
 This process may take +/- 48 hours.
 
 After that, you can update your `settings.xml` of your Maven:
 
-
-## Initialize the project
-
-All steps in this section are detailed in the [official documentation](http://central.sonatype.org/pages/apache-maven.html) and modify `pom.xml` of the project.
+Initialize the project: All steps in this section are detailed in the [official documentation](http://central.sonatype.org/pages/apache-maven.html) and modify `pom.xml` of the project.
```
Comment (surli) [inline doc/_release/Release.md]: Do we still need this part?

**Comment by monperrus** [comment]
thanks @surli, comments addressed, ready to merge

**Pull request merged**